### DISCUSSION TITLE
Questions of Quectel for OpenWrt router

### DISCUSSION THREAD

**Discussion by mglwolf** [IchthysMaranatha/asterisk-chan-quectel#56] [General]
Hi guys,
I'm going to upgrade my Huawei 3G dongle to Quectel 4G device, because 3G is going to retire in my country, my Asterisk server is running on OpenWrt device (MiR3P), which has a USB 3.0 port.
I have some questions:
1. what the current required, 600ma or much higher?
2. what about your experience of audio quality? for my huawei 3G not real good, choppy sometimes, I'm sure not network issue.
3. what dependencies for Quectel on OpenWrt?
4. can I use "chan_quectel3.5openwrtmips24kc" directly? or I need to build my own one? my hardware with MediaTek MT7621AT, OP 23.05.0.

Thanks

**Comment by mglwolf** [comment]
And which ECxx USB module you will recommend?

**Comment by IchthysMaranatha** [comment]
Hi, you can try using the binary directly with asterisk and alsalib installed if modem has been recognized and you can communicate with it via minicom. If it doesn't work, you'll need to use this sdk to build the binary
https://downloads.openwrt.org/releases/23.05.2/targets/ramips/mt7621/openwrt-sdk-23.05.2-ramips-mt7621_gcc-12.3.0_musl.Linux-x86_64.tar.xz

**Comment by mglwolf** [comment]
Hi ,
I tried chan_quectel3.5openwrtmips24kc, ramips-mt7620-openwrt-zbtwe826 on 2 openwrt devices, Quectel can be recognized, and I can see the incoming call in cli, but 2 issues, 

1error msg and can't make a call:
[Dec 28 12:56:50] WARNING[2758]: utils.c:2206 tvfix: warning too large timestamp 561393.0
[Dec 28 12:56:50] WARNING[2758]: utils.c:2206 tvfix: warning too large timestamp 99.1988876608

2 service of asterisk got kill for inbound call.

last could guide me to build my own one (chan_quectel.so)?  thank you!

**Comment by IchthysMaranatha** [comment]
Please try with this (with openwrt 22 if 23 doesn't work) https://github.com/IchthysMaranatha/asterisk-chan-quectel/files/11248976/chan_quectelOpenwrt22Ast18mips24kc.zip

**Comment by IchthysMaranatha** [comment]
Please try on we826 with openwrt 22

**Comment by mglwolf** [comment]
Yes, but I got an issue, outbound call disconnect very 31 seconds, strange! but there's no issue at all for inbound call, no limit and stable!
